code/game/machinery/washing_machine.dm

```diff
@@ -57,161 +103,170 @@
+					for(var/T in typesof(/obj/item/clothing/shoes))
+						var/obj/item/clothing/shoes/J = new T
+						if((wash_color == J.item_color) && J.dyeable)
+							qdel(O)
+							newcontents += J
+							break
+				if(istype(O, /obj/item/clothing/mask/bandana)) //bandanas
+					for(var/T in typesof(/obj/item/clothing/mask/bandana))
+						var/obj/item/clothing/mask/bandana/J = new T
+						if((wash_color == J.item_color) && J.dyeable)
+							qdel(O)
+							newcontents += J
+							break
+				if(istype(O, /obj/item/clothing/head/soft)) // soft-caps
+					for(var/T in typesof(/obj/item/clothing/head/soft))
+						var/obj/item/clothing/head/soft/J = new T
+						if((wash_color == J.item_color) && J.dyeable)
+							qdel(O)
+							newcontents += J
+							break
+				if(istype(O, /obj/item/clothing/head/beanie)) // non-striped beanies
+					for(var/T in typesof(/obj/item/clothing/head/beanie))
+						var/obj/item/clothing/head/beanie/J = new T
+						if((wash_color == J.item_color) && J.dyeable)
+							qdel(O)
+							newcontents += J
+							break
+				if(istype(O, /obj/item/clothing/accessory/scarf)) // scarves
+					for(var/T in typesof(/obj/item/clothing/accessory/scarf))
+						var/obj/item/clothing/accessory/scarf/J = new T
+						if((wash_color == J.item_color) && J.dyeable)
+							qdel(O)
+							newcontents += J
+							break
+				if(istype(O, /obj/item/bedsheet)) // bedsheets
+					for(var/T in typesof(/obj/item/bedsheet))
+						var/obj/item/bedsheet/J = new T
+						if(wash_color == J.item_color)
+							qdel(O)
+							newcontents += J
+							break
+				if(istype(O, /obj/item/storage/wallet/color)) // only for wallets from the arcade
+					for(var/T in typesof(/obj/item/storage/wallet/color))
+						var/obj/item/storage/wallet/color/J = new T
+						if(wash_color == J.item_color)
+							qdel(O)
+							newcontents += J
+							break
+				if(istype(O, /obj/item/toy/carpplushie)) // carp plushies! was gonna add for fox plushies but their pathing makes it not an option currently
+					for(var/T in typesof(/obj/item/toy/carpplushie))
+						var/obj/item/toy/carpplushie/J = new T
+						if(wash_color == J.item_color)
+							qdel(O)
+							newcontents += J
+							break
+			for(var/I in 1 to (length(newcontents)))
+				contents += newcontents[I]
 		QDEL_NULL(crayon)
 
-
-	if( locate(/mob,contents) )
-		state = 7
-		gibs_ready = 1
+	running = FALSE
+	if(!full)
+		bloody = FALSE
 	else
-		state = 4
+		playsound(src, 'sound/weapons/jug_filled_impact.ogg', 25)
 	update_icon()
-
-/obj/machinery/washing_machine/verb/climb_out()
-	set name = "Climb out"
-	set category = "Object"
-	set src in usr.loc
-
-	sleep(20)
-	if(state in list(1,3,6) )
-		usr.loc = src.loc
-
+	sleep(5)
+	if(stat & NOPOWER)
+		playsound(src, 'sound/machines/terminal_off.ogg', 45)
+	else
+		playsound(src, 'sound/machines/defib_success.ogg', 25)
 
 /obj/machinery/washing_machine/update_icon()
-	icon_state = "wm_[state][panel]"
+	if(bloody)
+		if(running)
+			state = 8
```

Comment:
I saw that you were mostly replacing these states which is *lovely* since they're nasty

Reply:
Yep, only keeping them around (for now), for icon states, if I find a better solution I'll switch to that

Reply:
Only keeping them around for easier icon_states, if I find a better solution I'll probably toss it the states